python/pyspark/sql/tests.py

```diff
@@ -901,20 +901,43 @@ def __call__(self, x):
         self.assertEqual(return_type, f_.returnType)
 
     def test_stopiteration_in_udf(self):
-        # test for SPARK-23754
-        from pyspark.sql.functions import udf
+        from pyspark.sql.functions import udf, pandas_udf, PandasUDFType
         from py4j.protocol import Py4JJavaError
 
+        def do_test(action, *args, **kwargs):
```

Comment:
Likewise, let's just do

```python
exc_message = "Caught StopIteration thrown from user's code; failing the task"

with self.assertRaisesRegexp(Py4JJavaError, exc_message):
    ... pandas_udf ...
```

I think it doesn't particularly reduce the codes.